rust/agama-lib/src/localization/store.rs

```diff
@@ -0,0 +1,51 @@
+//! Implements the store for the localization settings.
+// TODO: for an overview see crate::store (?)
+
+use super::{LocalizationClient, LocalizationSettings};
+use crate::error::ServiceError;
+use zbus::Connection;
+
+/// Loads and stores the storage settings from/to the D-Bus service.
+pub struct LocalizationStore<'a> {
+    localization_client: LocalizationClient<'a>,
+}
+
+impl<'a> LocalizationStore<'a> {
+    pub async fn new(connection: Connection) -> Result<LocalizationStore<'a>, ServiceError> {
+        Ok(Self {
+            localization_client: LocalizationClient::new(connection).await?,
+        })
+    }
+
+    pub async fn load(&self) -> Result<LocalizationSettings, ServiceError> {
+        // TODO: we should use a single D-Bus call with Properties.GetAll
+        // but LocaleProxy does not have it, only get_property for individual methods
+        // and properties_proxy is private
```

Comment:
Just use Property proxy instead of LocaleProxy for given object.
https://docs.rs/zbus/latest/zbus/fdo/struct.PropertiesProxy.html

Reply:
so basically you can construct proxy yourself...but it is usually not necessary as it is cached and zbus checking PropertiesChange signal for you, so in the end it is very efficient if property does not change too much.